private/bufpkg/bufconnect/interceptors.go

```diff
@@ -43,64 +44,120 @@ func NewSetCLIVersionInterceptor(version string) connect.UnaryInterceptorFunc {
 }
 
 // NewAuthorizationInterceptorProvider returns a new provider function which, when invoked, returns an interceptor
-// which will look up an auth token by address and set it into the request header.  This is used for registry providers
-// where the token is looked up by the client address at the time of client construction (i.e. for clients where a
-// user is already authenticated and the token is stored in .netrc)
+// which will set the auth token into the request header by the provided option.
 //
 // Note that the interceptor returned from this provider is always applied LAST in the series of interceptors added to
 // a client.
-func NewAuthorizationInterceptorProvider(container app.EnvContainer) func(string) connect.UnaryInterceptorFunc {
+func NewAuthorizationInterceptorProvider(option SetAuthTokenOption) func(string) connect.UnaryInterceptorFunc {
```

Comment:
Options are generally things that are optional in constructors, whereas this appears to be a required parameter.

Reply:
This option is required as we must either set the auth token by provided token or container.

Reply:
See comment below - I am not quite sure why this is preferred over having two constructors

Reply:
Blocking the merging of this PR until this is resolved - I think the previous style, with two constructors, is correct, and there should be no new type named *Option introduced to this PR with one constructor deleted

Reply:
The constructor is taking tokenset to provide the auth token in the most recent commit